src/ctl.c

```diff
@@ -2980,6 +2982,55 @@ arena_i_retain_grow_limit_ctl(tsd_t *tsd, const size_t *mib,
 	return ret;
 }
 
+static int
+arena_i_name_ctl(tsd_t *tsd, const size_t *mib, size_t miblen,
+    void *oldp, size_t *oldlenp, void *newp, size_t newlen) {
+	int ret;
+	unsigned arena_ind;
+	const char *name = NULL;
+
+	READ_XOR_WRITE();
+
+	malloc_mutex_lock(tsd_tsdn(tsd), &ctl_mtx);
+	MIB_UNSIGNED(arena_ind, 1);
+	if (arena_ind == MALLCTL_ARENAS_ALL || arena_ind >=
+	    ctl_arenas->narenas) {
+		malloc_printf("Invalid arena index %d. This function only takes in "
+		"indices no larger than %d.\n", arena_ind, ctl_arenas->narenas);
+		ret = EINVAL;
+		goto label_return;
+	}
+	arena_t *arena = arena_get(tsd_tsdn(tsd), arena_ind, false);
+	if (arena == NULL) {
+		ret = EFAULT;
+		goto label_return;
+	}
+	if (oldp != NULL && oldlenp != NULL) {
+		/*
+		 * Read the arena name.  When reading, the input oldp must point
+		 * to an array with a length no shorter than ARENA_NAME_LEN.
+		 */
+		name = arena_name_get(arena);
+		if (name == NULL) {
```

Comment:
This won't be possible, unless some bug / corruption happened, right? In that case, do an assert would be fine.

Reply:
Yes, assertion removed.